content/news/buzz/2025-curtin-capstone-semester-end-blog/contents.lr

```diff
@@ -0,0 +1,102 @@
+
+Our primary goals moving forward include:
+
+### Toga Web Testing
+
+Toga does not currently support testing for the web platform. This is because when Toga runs tests for the other platforms, it starts a testbed application with the Pytest test suite running as a thread. Additionally, since the testbed app and test suite are running in the same process, they share the same memory space. The test suite is built assuming this is the case as it creates Toga objects, then adds them directly to the testbed app, later modifying them at the API level to then test that the platform being tested has rendered the changes as expected.
+
+This cannot happen with Toga web apps, as they do not support multithreading due to WebAssembly only running single-threaded. Therefore, we have to find a way for the test suite to run correctly when it is running in its own process, separate from the testbed app. Our goals are to prove-out a general approach that will allow this to occur, which will include proof-of-concepts that show it is feasible and provide some initial specifications and guidelines for future pieces of work.
+
+Our current idea is to ‘replace’ any Toga objects created in the test suite with proxy objects, which will act as a ‘wrapper’ around its counterpart in the remote testbed app. This will involve remotely creating the object in the testbed app and forwarding commands to modify its attributes and retrieve data from it. This will allow the testbed app to render any changes the test suite is expecting so that the test suite can test the web app at the API as well as the platform level.
+
+Speaking of testing the testbed app at the platform level, we need a way for the test suite to test that any modifications at the API level are correctly rendered in-app. This is where Playwright will come in. Playwright allows us to inspect the testbed web app at the DOM level and verify that its state matches the expectations of the test suite.
+
+Lastly, we want to have a look at how Briefcase could run the testbed app and the Pytest test suite as two separate processes, while still allowing the collection and reporting of test suite results and code coverage results.
+
+For more information, the discussion leading up to this point can be found in our [Discussion Post on Toga](https://github.com/beeware/toga/discussions/3440). The corresponding ticket to this part of our capstone project can be found in our [Issue Ticket](https://github.com/beeware/toga/issues/3545).
+
+### PyScript Briefcase and Toga Dependencies
+
+During Mitchell’s development of the DateInput Web Widget, we discovered some issues resulting from create_proxy. This issue was resolved separately, but led to discussions regarding ownership of Pyscript, as well as Shoelace and Bootstrap, within the wider Beeware architecture. Currently, the Briefcase Web Template inserts these into index.html, although these should be dependencies of Toga.
```

Comment:
Markup and project name tweak; but "some issues resulting from create_proxy" is inside baseball unless you know the internals. "some issues resulting from how PyScript APIs were being used" might be better:
```suggestion
During Mitchell’s development of the DateInput Web Widget, we discovered some issues resulting from ``create_proxy``. This issue was resolved separately, but led to discussions regarding ownership of PyScript, as well as Shoelace and Bootstrap, within the wider BeeWare architecture. Currently, the Briefcase Web Template inserts these into index.html, although these should be dependencies of Toga.
```